Week1 [STEP3] 김마루

marukim365 opened the pull request:
feat: 빼빼로 그리기 코드 및 순서도 작성

Week1 STEP3 주문에 따른  빼빼로 그리기
 - 요구사항에 맞게 총 4개의 함수로 코드 작성
 - 함수의 흐름에 따른 순서도 작성

marukim365 committed:
STEP2.Draw ice cream & edit README.md

marukim365 committed:
edit function, draw flowchart

marukim365 committed:
feat: 빼빼로 그리기 코드 및 순서도 작성

Week1 STEP3 주문에 따른  빼빼로 그리기
 - 요구사항에 맞게 총 4개의 함수로 코드 작성
 - 함수의 흐름에 따른 순서도 작성

FirstDo reviewed (changes requested):
마루님 STEP3도 고생하셨습니다!!!
깔끔하게 잘해주셔서 저도 너무 좋고 크게 피드백 드릴 부분이 없었습니다 ㅎㅎ

몇가지 코맨트와 부탁드리고 싶은점을 적었으니 확인하고 수정해주세요 :)
(저와 의견이 다르다면 댓글로 의견을 남겨주셔도 됩니다!)

마루님이 깔끔하게 잘해주셔서 저도 신나서 열심히 피드백을 드리다보니, 프로젝트 요구사항을 넘어선 부분이 있을수도 있습니다!
해당부분 수정이 힘들거나 굳이 할필요 없다고 생각되시면 굳이 수정하지 않으셔도 괜찮습니다

## 이번스텝의 핵심 요구사항은 아래와 같습니다
- [x] 함수를 나눠서 작성하기
- [x]  다양한 빼빼로를 출력해보기
- [x]  논리적으로 순서도 작성하기

## 드리고 싶은 말씀

### 1. 커밋을 작게 끊어서 해주세요

커밋을 찍는이유가 뭘까요?
모든 로직을 다 작성한후 커밋을 하면 나중에  특정부분에 문제가 생겨서 해당 커밋이전으로 돌아가야할때 전체를 다 돌려야 합니다.
기능단위로 끊어서 (여기서는 함수) 커밋하는걸 추천합니다.
이번에 수정할때도 마찬가지입니다!

ex)
feat: drawHolder 함수 구현
feat: drawPepero 함수 구현
add: 순서도 추가

### 2. PR 내용을 조금만 더 자세히 써주세요

`Pull Request 보낼때 꼭 참고하세요!` 를 참고해주세요

추가로 특히 이번과제처럼 어떤 print 결과를 보여줘야 한다면
PR문 마지막에 해당결과(모든 빼빼로 출력 결과)까지 첨부하는게 좋습니다
지금은 어차피 제가 마루님 코드를 다 뜰어보고 돌려보지만,  현업에서는 그러기 힘들겠죠?

PR을 보낼때, PR본문에서는 어떤일을 어떻게 했는지 동료들에게 잘 설명해야 합니다
현재 본문만 보고는 마루님이 어떤점을 신경쓰셨는지, 어떤 고민이 있으셨는지 제가 파악하기가 쉽지 않습니다..!

marukim365 committed:
feat: drawPepero.swift 파일 생성

기존 main.swift에 작성된 코드에서 새롭게 drawPepero.swift 파일을 생성하여 코드를 분리함

marukim365 committed:
docs: 순서도 파일 수정

기존에는 자세하게 나와있던 인자값을 간단하게 파라미터 이름으로 수정함

marukim365 committed:
style: 코드 스타일 수정

- 함수명을 choiceBodyAndTopping에서 fetchBodyAndTopping으로 수정함
- drawPepero 함수 매개변수 순서를 출력순서와 동일하게 변경함
- drawBody 함수 매개변수 명을 변경함, body -> bodyWithToppings

FirstDo committed:
Merge branch 'ss_11_marukim' into ss_11_marukim
